### DISCUSSION TITLE
Create column with a non-database table

### DISCUSSION THREAD

**Discussion by DavidGilWeb** [filamentphp/filament#11556] [Help]
### Package

Table builder

### Package Version

v3.0.0

### How can we help you?

Hi, I am trying to create a column in the table, which displays data from an array. I return this array from a helper, but it seems that if it is not in database, I can't display it, how could I do it without creating a database table?

**Comment by zeeshantariq08** [comment]
You can make an attribute accessor 
  `  
  ```
 protected function firstName(): Attribute
        {
            return Attribute::make(
                get: fn (string $value) => ucfirst($value),
            );
        }
```
`

**Reply by DavidGilWeb** [reply]
Thank you for your response first and foremost.
My idea is to be able to display the data in the resource table, should I add this function there?, and if so, how should I call it in the table column?

**Reply by leandrocfe** [reply]
What is your array structure? How would you like to display it on the table?

**Reply by DavidGilWeb** [reply]
My structure is like this, I can make it associative if needed, what I would like is to show a list with all the versions.
array (size=1)
  0 => string '0.0.1' (length=5)
My goal is to be able to see all the version folders so I can list them.

**Reply by DavidGilWeb** [reply]
Otherwise, my idea is to do it simply with a laravel view and controller, but I wanted it to follow the same structure of filament resources.

**Reply by leandrocfe** [reply]
You can use `->state()`
https://filamentphp.com/docs/3.x/infolists/entries/getting-started#calculated-state

```php
Tables\Columns\TextColumn::make('versions')
    ->state(['0.0.1', '0.0.2'])
    ->badge()
```

**Reply by DavidGilWeb** [reply]
Finally, I managed to do it by creating a few simple pages, without the need to create resources, thank you both for your help.

**Comment by DavidGilWeb** [comment]
Create pages with 'php artisan make:filament-page NamePage'.
Add in Filament/Page/Page.php functionality and display data in Resources/Views/Filament/Pages/page.blade.php
